src/stores/Store.ts

```diff
@@ -74,6 +81,246 @@ export default class Store {
+			if (this.selectedActBox && this.selectedService && this.selectedMethod) {
+				const editedMessage: ActMessageItem = {
+					actBox: this.selectedActBox,
+					fullServiceName: this.selectedService,
+					methodName: this.selectedMethod?.methodName,
+					message: this.editorCode,
+					delay: this.editedMessageSendDelay,
+				};
+				return editedMessage;
+			}
+			return undefined;
+		}
+		return undefined;
+	};
+
+	@action saveEditedMessage = () => {
+		if (this.editedMessageIndex >= 0) {
+			const message: ParsedMessageItem | ActMessageItem | undefined =				this.buildEditedMessage();
+			if (message !== undefined) {
+				if (this.selectedSchemaType === 'parsed-message' && isParsedMessageItem(message)) {
+					this.parsedMessagesHistory[this.editedMessageIndex] = message;
+				} else if (this.selectedSchemaType === 'act' && isActMessageItem(message)) {
+					this.actMessagesHistory[this.editedMessageIndex] = message;
+				}
+				this.indicators[this.editedMessageIndex] = 'indicator-edited';
+			}
+
+			this.setEditMessageMode(false);
+		}
+	};
+
+	@action setEditMessageMode = (mode: boolean) => {
+		this.editMessageMode = mode;
+		if (mode === false) {
+			this.setEditedMessageIndex(-1);
+		}
+		if (this.selectedSchemaType === 'parsed-message') {
+			localStorage.setItem('editParsedMessageMode', mode.toString());
+		} else if (this.selectedSchemaType === 'act') {
+			localStorage.setItem('editActMessageMode', mode.toString());
+		}
+	};
+
+	@action selectMessage = (index: number) => {
+		if (this.selectedSchemaType === 'parsed-message') {
+			this.setEditorProperties(
+				this.parsedMessagesHistory[index].sessionId,
+				this.parsedMessagesHistory[index].dictionary,
+				this.parsedMessagesHistory[index].messageType,
+				this.parsedMessagesHistory[index].message as string,
+			);
+		} else if (this.selectedSchemaType === 'act') {
+			this.setEditorProperties(
+				this.actMessagesHistory[index].actBox,
+				this.actMessagesHistory[index].fullServiceName,
+				this.actMessagesHistory[index].methodName,
+				this.actMessagesHistory[index].message as string,
+			);
+		}
+
+		this.setEditMessageMode(true);
+		this.setEditedMessageIndex(index);
+	};
+
+	setEditorProperties = (
+		sessionOrActBox: string | null,
+		dictionaryOrService: string | null,
+		messageTypeOrMethod: string | null,
+		editorCode: string,
+	) => {
+		if (this.selectedSchemaType === 'parsed-message') {
+			this.selectedSession = sessionOrActBox;
+			this.selectedDictionaryName = dictionaryOrService;
+			this.selectedMessageType = messageTypeOrMethod;
+		} else if (this.selectedSchemaType === 'act') {
+			this.selectedActBox = sessionOrActBox;
+			this.selectedService = dictionaryOrService;
+			this.setSelectedMethod(messageTypeOrMethod);
+		}
+
+		if (editorCode) {
+			this.setEditorCode(editorCode);
+		}
+	};
+
+	@action setEditorCode = (code: string) => {
+		this.editorCode = code;
+	};
+
+	@action getIndicatorByResult = (code: number): Indicator => {
```

Comment:
its not an action. it doesnt change state. this function too small to be a function and use only inside `replayMessage` function. move function declaration there or use it inline